### PR TITLE
Simplify $ transpiling and fix newline character bug

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR addresses an issue where `lineTerminatorMatcher(excludeCRLF = true)` returns an empty Regex AST, which could lead to incorrect behavior when passed unchecked to RegexRepetition. 

This update also improves handling of end-of-line characters following a line anchor (e.g., `$\r`, `$\u2028`), falling back to CPU due to cuDF's lack of support for negative lookahead. However, [checkUnsupported](https://github.com/NVIDIA/spark-rapids/blob/61acf56de3fdf71e3c6c0cedde0939b138211818/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala#L979) will already catch these cases before reaching this point.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala [hunk 1]
```diff
@@ -1350,45 +1336,23 @@ class CudfRegexTranspiler(mode: RegexMode) {
...
-                        RegexChar('$')))
                     popBackrefIfNecessary(capture)
                   case RegexCharacterClass(true, parts)
                       if parts.forall(!isBeginOrEndLineAnchor(_)) =>
-                    r(j) = RegexSequence(ListBuffer(
-                      lineTerminatorMatcher(excludeCRLF = true, capture = false),
-                        RegexChar('$')))
                     popBackrefIfNecessary(false)
-                  case RegexChar(ch) if ch == '\n' =>
+                  case RegexChar(ch) if "\n\r\u0085\u2028\u2029".contains(ch) =>
```
Comment (revans2) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala:1350]: is this lineTerminatorChars, but concated together? Could we make this unified? For both maintainability but also to make it more readable?

File: tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionSuite.scala [hunk 2]
```diff
@@ -150,4 +150,25 @@ class RegularExpressionSuite extends SparkQueryCompareTestSuite {
     Charset.defaultCharset().name() == "UTF-8"
   }
 
+  private def extractStringsLineEnd(session: SparkSession): DataFrame = {
+    import session.sqlContext.implicits._
+    Seq[(String)](
+      ("123\n"),
```
Comment (revans2) [inline tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionSuite.scala:156]: Can we extend this to also include some of the other lineTerminatorChars characters?

**Review by revans2 — DISMISSED** [review]
Just some nits

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala (hunk 1 above)
Comment (SurajAralihalli) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala]: Done, thanks for pointing it out!

**Review by revans2 — DISMISSED** [review]
This looks good to me, but I really want @NVnavkumar to also take a look. I am not as much of an expert as he is.

**Review by NVnavkumar — APPROVED** [review]
LGTM

**Comment by NVnavkumar** [comment]
build

**Comment by SurajAralihalli** [comment]
build

**Comment by SurajAralihalli** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**